segment_point_cloud_topic: "/depth_segmentation_node/object_segment"
world_frame_id: "map"
publish_feature_blocks_marker: false
use_label_propagation: true

voxblox:
  voxel_size: 0.02
  voxels_per_side: 8
  truncation_distance_factor: 3.0
  voxel_carving_enabled: false
  max_ray_length_m: 3.5

gsm:
  min_label_voxel_count: 20
  label_propagation_td_factor: 1.0

pairwise_confidence_merging:
  enable_pairwise_confidence_merging: true
  merging_min_overlap_ratio: 0.1
  merging_min_frame_count: 2

semantic_instance_segmentation:
  enable_semantic_instance_segmentation: false

meshing:
  visualize: false
  update_mesh_every_n_sec: 0.0
  publish_segment_mesh: false
  publish_scene_mesh: true
  mesh_filename: "mesh.ply"
  compute_and_publish_bbox: false
  min_weight: 3

mesh_config:
  min_weight: 3

icp:
  enable_icp: false
  keep_track_of_icp_correction: true

object_database:
  publish_gsm_updates: false
  max_segment_age: 10
  scene_gsm_update_topic: "gsm_update"
  segment_gsm_update_topic: "gsm_update"
  validate_merged_object: "validate_merged_object"

sliding_window:
  is_on: false
  radius_m: 1.0
  imu_frame_id: imu

  # Between 0 and 1. As soon as the distance between sliding window and camera
  # is larger than this fraction times the window radius, the window is updated
  # and its position is set equal the camera position.
  update_fraction: 0.5
